File: web/src/comm/validater.js
const validate = {
    isPhone: phone => {
        let pattern = /^[1][3,4,5,7,8][0-9]{9}$/;
        return pattern.test(phone) ? true : false;
    },
    isEmail: email => {
        let pattern = /^(\w-*\.*)+@(\w-?)+(\.\w{2,})+$/;
        return pattern.test(email) ? true : false;
    },
    limitName: name => {
        let pattern = /^[a-zA-Z0-9_\u4E00-\u9FA5]{3,10}$/;
        return pattern.test(name) ? true : false;
    }
};
export {
    validate
}

File: server/validater.py
# -*- coding: utf-8 -*-
# @Author: wangfpp
# @Date:   2018-11-22 18:21:35
# @Last Modified by:   wangfpp
# @Last Modified time: 2018-11-22 18:22:05
import re
class validate():
    def __init__(self):
        self.emainre = re.compile(r'[^\._][\w\._-]+@(?:[A-Za-z0-9]+\.)+[A-Za-z]+$')
        self.phonere = re.compile(r'[1][^1269]\d{9}')
    def isEmail(self, email):
        res = self.emainre.match(email)
        if res:
            return True
        else:
            return False
    def isPhone(self, phone):
        res = self.phonere.match(phone)
        if res:
            return True
        else:
            return False
if __name__ == "__main__":
    a = validate()
